api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -351,141 +351,161 @@ const (
+	// TODO(alberto): Signal this in a condition.
+	// When sshKey is set, the controllers will generate a machineConfig with the sshAuthorizedKeys https://coreos.github.io/ignition/configuration-v3_2/ populated with this value.
+	// This MachineConfig will be part of every payload generated by the controllers for any NodePool of the HostedCluster.
+	// Changing this value will trigger a rollout for all existing NodePools in the cluster.
+	// +optional
 	// +immutable
 	SSHKey corev1.LocalObjectReference `json:"sshKey"`
 
-	// IssuerURL is an OIDC issuer URL which is used as the issuer in all
-	// ServiceAccount tokens generated by the control plane API server. The
-	// default value is kubernetes.default.svc, which only works for in-cluster
+	// issuerURL is an OIDC issuer URL which will be used as the issuer in all
+	// ServiceAccount tokens generated by the control plane API server via --service-account-issuer kube api server flag.
+	// https://k8s-docs.netlify.app/en/docs/reference/command-line-tools-reference/kube-apiserver/
+	// https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#serviceaccount-token-volume-projection
+	// The default value is kubernetes.default.svc, which only works for in-cluster
 	// validation.
-	//
+	// If the platform is AWS and this value is set, the controller will update an s3 object with the appropriate OIDC documents (using the serviceAccountSigningKey info) to into that issuerURL.
+	// The expectation is for this s3 url to be backed by an OIDC provider in the AWS IAM.
 	// +kubebuilder:default:="https://kubernetes.default.svc"
 	// +immutable
 	// +optional
-	// +kubebuilder:validation:Format=uri
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="omitempty is immutable"
+	// +kubebuilder:validation:XValidation:rule="isURL(self)",message="issuerURL must be a valid absolute URL"
 	IssuerURL string `json:"issuerURL,omitempty"`
 
-	// ServiceAccountSigningKey is a reference to a secret containing the private key
-	// used by the service account token issuer. The secret is expected to contain
-	// a single key named "key". If not specified, a service account signing key will
-	// be generated automatically for the cluster. When specifying a service account
-	// signing key, a IssuerURL must also be specified.
+	// ServiceAccountSigningKey is a local reference to a secret that must have a "key" key whose content must be the private key
+	// used by the service account token issuer.
+	// If not specified, a service account signing key will
+	// be generated automatically for the cluster.
+	// When specifying a service account signing key, an IssuerURL must also be specified.
+	// If the reference is set but none of the above requirements are met, the HostedCluster will enter a degraded state.
+	// TODO(alberto): Signal this in a condition.
 	//
 	// +immutable
 	// +kubebuilder:validation:Optional
```

Comment:
Does this need to be removed? I see the one on `Configuration` below was removed.

Reply:
consolidating on using 	// +optional everywhere as we go